.github/workflows/publish-swagger-hub.yml

```
@@ -0,0 +1,54 @@
+name: "Publish OpenAPI to Swaggerhub"
+
+on:
+  workflow_call:
+    inputs:
+      version:
+        required: true
+        description: Version that will be published to Swaggerhub
+        type: string
+
+jobs:
+  swagger-api:
+    runs-on: ubuntu-latest
+    env:
+      SWAGGERHUB_API_KEY: ${{ secrets.SWAGGERHUB_API_KEY }}
+      SWAGGERHUB_USER: ${{ secrets.SWAGGERHUB_USER }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v4
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+
+      - name: Setup node
+        uses: actions/setup-node@v4
+
+      - name: Install Swagger CLI
+        run: |
+          npm i -g swaggerhub-cli
+
+      - name: Extract versions
+        run: |
+          export DOWNSTREAM_VERSION="${{ inputs.version }}"
+          echo "DOWNSTREAM_VERSION=${DOWNSTREAM_VERSION}" >> "$GITHUB_ENV"
+
+      # create API, will fail if exists
```

Not necessary to extract and put to envs. You can set it in the env of the job
```suggestion
          npm i -g swaggerhub-cli

      # create API, will fail if exists
```